controllers/items.go

```diff
@@ -0,0 +1,55 @@
+package controllers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"go-rest-api/services"
+
+	"github.com/gorilla/mux"
+)
+
+func GetItems(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	items, err := services.GetItems()
+	if err != nil {
+		errorResponse(w, "Some Error Occurred", 500)
+		return
+	}
+
+	json.NewEncoder(w).Encode(items)
+}
+
+func GetItem(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	vars := mux.Vars(r)
+	taskID, err := strconv.Atoi(vars["id"])
+	if err != nil {
+		errorResponse(w, "ID path parameter must be an Integer", http.StatusBadRequest)
+		return
+	}
+
+	item, err := services.GetItem(taskID)
+	if err != nil {
+		log.Println("log error: ", err.Error())
+		if err.Error() == "404" {
```

Comment:
Continuing my comment for this particular error, if you define in your service package a const like `ItemNotFoundError = errors.New("item not found")`, you could use it here like `if err == service.ItemNotFoundError` and avoid hardcoding the message throughout packages